From ea5ac2128a5a896619d42699be061e25ab49f586 Mon Sep 17 00:00:00 2001
From: Hansong Zhang <hsz@google.com>
Date: Fri, 11 Dec 2020 11:56:15 -0800
Subject: [PATCH 11/12] [Q_asb_2021-04] avrc_copy_packet: Zero initialize
 packet

Bug: 174149901
Change-Id: Iefa41749ebbacd34afaa24131de7ee25d706e23f
Tag: Security
Test: POC
(cherry picked from commit 960c3f3c9a1f912544b92b7a744e22069a0bc27e)
(cherry picked from commit 0f447e4fffc45b10d29e1cfcc2eb66067abaf034)
---
 stack/avrc/avrc_api.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/stack/avrc/avrc_api.cc b/stack/avrc/avrc_api.cc
index e09a9a279..d557782ff 100644
--- a/stack/avrc/avrc_api.cc
+++ b/stack/avrc/avrc_api.cc
@@ -260,7 +260,7 @@ static uint8_t avrc_get_packet_type(BT_HDR *pp_pkt) {
 static BT_HDR* avrc_copy_packet(BT_HDR* p_pkt, int rsp_pkt_len) {
   const int offset = MAX(AVCT_MSG_OFFSET, p_pkt->offset);
   const int pkt_len = MAX(rsp_pkt_len, p_pkt->len);
-  BT_HDR* p_pkt_copy = (BT_HDR*)osi_malloc(BT_HDR_SIZE + offset + pkt_len);
+  BT_HDR* p_pkt_copy = (BT_HDR*)osi_calloc(BT_HDR_SIZE + offset + pkt_len);
 
   /* Copy the packet header, set the new offset, and copy the payload */
   memcpy(p_pkt_copy, p_pkt, BT_HDR_SIZE);
-- 
2.34.1

